### PR TITLE
Fix Windows build

### PR DESCRIPTION
**Pull request by as-cii**
This should unblock using AppVeyor on atom/atom#12170.

The main issue with the previous scripts was that we were building the URL for downloading Node x86 for Windows in a wrong way, which was preventing it from being downloaded. As part of fixing that, we have also restructured the scripts a little bit to make them more similar to their Mac/Linux counterparts.

Please, note that we have disabled running tests because some of them appear to be written in a platform-specific way. I think fixing them can be deferred to another pull-request, so that we can start using the new version of apm on Atom right away. /cc: @damieng 

/cc: @nathansobo @atom/core

### CODE REVIEW

**Comment by as-cii** [comment]
Before merging this, I am going to publish a pre-release version so that we can test it on Atom.

**Comment by as-cii** [comment]
After upgrading the dependency on Atom, I confirm [this makes the build green](https://ci.appveyor.com/project/Atom/atom/build/1984). 💚

File: bin/npm.cmd [hunk 1]
```diff
@@ -1,17 +1,6 @@
 @echo off
 setlocal EnableDelayedExpansion
 
-:: Try to find git.exe in path
```
Comment (damieng) [inline bin/npm.cmd]: Why is this removed?  It was added so that people on Windows using APM on the command line installing packages that look for git.exe could actually find the one installed by Git Desktop because it isn't in the path...

It shouldn't be interfering with AppVeyor builds in any way.

Comment (nathansobo) [inline bin/npm.cmd]: We didn't understand why it was needed as part of `npm.cmd`... Made more sense in `apm.cmd`. Assumed it was copied and pasted. Am I wrong?

Comment (damieng) [inline bin/npm.cmd]: Are there not scenarios where atom or a user might fire of this npm.cmd
without going via apm.cmd?

Comment (as-cii) [inline bin/npm.cmd]: > Are there not scenarios where atom or a user might fire of this npm.cmd
> without going via apm.cmd?

As far as Atom is concerned I am pretty sure we never invoke `npm` directly. I think the same holds true for users, because they'll interface with `npm` transparently via `apm`. Maybe I am missing something, though? Did you have anything specific in mind, @damieng?

**Comment by as-cii** [comment]
I'll proceed with merging and publishing this. We can refine it later if there's any concern before we roll the railcars on atom/atom.

Thanks @nathansobo and @damieng!

**Comment by nathansobo** [comment]
⚡️
